rust/chains/hyperlane-cosmos/src/libs/address.rs

```diff
@@ -0,0 +1,53 @@
+use std::str::FromStr;
+
+use cosmrs::{crypto::secp256k1::SigningKey, AccountId};
+use derive_new::new;
+use hyperlane_core::{ChainResult, H256};
+
+/// decode bech32 address to H256
+pub fn bech32_decode(addr: String) -> ChainResult<H256> {
+    let account_id = AccountId::from_str(&addr)?;
+
+    // although `from_slice` can panic if the slice is not 32 bytes long,
+    // we know that we're passing in a value that is 32 bytes long because it was decoded from
+    // bech32
+    Ok(H256::from_slice(&account_id.to_bytes()))
+}
+
+/// Wrapper around the cosmrs AccountId type that abstracts keypair conversions and
+/// bech32 encoding
+#[derive(new)]
+pub struct CosmosAddress {
```

Comment:
may be a good place to have some quick unit tests if we're uncertain about the encoding / decoding like iiuc you flag in the PR description